### PR TITLE
Basic YAML generation

### PR DESCRIPTION
**Pull request by porras**
It's incomplete but I wanted to share it early to gather feedback on the interest on merging it, and on the approach (of writing it in Crystal instead of wrapping libyaml like the parser does).

Things missing:

* Support for custom classes serialization macros
* Maybe more core classes
* Probably lots of edge cases (escaping strings, multiline paragraphs, etc.)

### CODE REVIEW

**Comment by sdogruyol** [comment]
:+1: from me

**Comment by MakeNowJust** [comment]
:+1: I want it to generate diagnostics in [tap](https://github.com/MakeNowJust/tap).

File: src/yaml/to_yaml.cr [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+class String
+  def to_yaml(yaml : YAML::Generator)
+    yaml << self
+  end
+end
+
+struct Number
+  def to_yaml(yaml : YAML::Generator)
+    yaml << self.to_s
```
Comment (asterite) [inline src/yaml/to_yaml.cr]: This `to_s` call is not necessary

File: src/yaml/to_yaml.cr [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+struct Bool
+  def to_yaml(yaml : YAML::Generator)
+    yaml << to_s
+  end
+end
+
+struct Set
+  def to_yaml(yaml : YAML::Generator)
+    to_a.to_yaml(yaml)
```
Comment (asterite) [inline src/yaml/to_yaml.cr]: I'd maybe write directly to the yaml generator and avoid creating an intermediate array

**Comment by asterite** [comment]
Looks good so far :-)

**Comment by porras** [comment]
I just removed the unnecessary `#to_s` and `#to_a`. I guess there's a potential refactor into a `YAML::Generator#array` method that yields some kind of `ArrayBuilder` (similar to the one in the JSON library) to avoid the very similar code for `Set`, `Array` and `Tuple`, but I think it'd add more complexity than it'd remove.

Should I rebase these commits together?

**Comment by asterite** [comment]
@porras Yes, please rebase/squash and then I'll merge it :-)

**Comment by porras** [comment]
@asterite I just did, thanks!

**Comment by asterite** [comment]
Thanks!! :-)

**Pull request merged**